src/clj/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_core.clj

```diff
@@ -87,55 +92,56 @@
    watcher :- (schema/protocol Watcher)]
   (let [dir-create? (fn [event]
                       (and (= :create (:type event))
-                           (fs/directory? (:path event))))]
+                           (fs/directory? (:full-path event))))]
     (DirWatchUtils/registerRecursive (:watch-service watcher)
                                      (->> events
                                           (filter dir-create?)
-                                          (map #(.toPath (:path %)))))))
+                                          (map #(.toPath (:full-path %)))))))
 
-(schema/defn retrieve-events :- {WatchKey [WatchEvent]}
+(defn get-event-maps-from-key
+  [watch-key]
+  (let [events (.pollEvents watch-key)]
```

Comment:
Should this be `if-let` or do we want to always map even if there are no events?